code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm

```diff
@@ -94,9 +85,118 @@
 	emote_see = list("honks", "bites into the banana", "plucks a banana off its head", "photosynthesizes")
 	maxHealth = 120
 	health = 120
-	speed = -10
+	speed = -1
 	loot = list(/obj/item/clothing/mask/gas/clown_hat, /obj/effect/gibspawner/human, /obj/item/soap, /obj/item/seeds/banana)
-	banana_time = 20
+	//Our peel dropping ability
+	var/datum/action/cooldown/rustle/banana_rustle
+	///Our banana bunch spawning ability
+	var/datum/action/cooldown/exquisite_bunch/banana_bunch
+
+/mob/living/simple_animal/hostile/retaliate/clown/banana/Initialize(mapload)
+	. = ..()
+	banana_rustle = new()
+	banana_rustle.Grant(src)
+	banana_bunch = new()
+	banana_bunch.Grant(src)
+
+///drops peels around the mob when activated
+/datum/action/cooldown/rustle
+	name = "Rustle"
+	desc = "Shake loose a few banana peels."
+	///which type of peel to spawn
+	var/banana_type = /obj/item/grown/bananapeel
+	///How many peels to spawn
+	var/peel_amount = 3
+	cooldown_time = 8 SECONDS
+	button_icon_state = "rustle"
+	icon_icon = 'icons/mob/actions/actions_clown.dmi'
+	background_icon_state = "bg_nature"
+
+/datum/action/cooldown/rustle/Activate(atom/target)
+	. = ..()
+	var/list/reachable_turfs = list()
+	for(var/turf/adjacent_turf in RANGE_TURFS(1, owner.loc))
+		if(adjacent_turf == owner.loc || !owner.CanReach(adjacent_turf) || !isopenturf(adjacent_turf))
+			continue
+		reachable_turfs += adjacent_turf
+
+	var/peels_to_spawn = min(peel_amount, reachable_turfs.len)
+	for(var/i in 1 to peels_to_spawn)
+		new banana_type(pick_n_take(reachable_turfs))
+	playsound(owner, 'sound/creatures/clown/clownana_rustle.ogg', 100)
+	animate(owner, time =1, pixel_x = 6, easing = CUBIC_EASING | EASE_OUT)
+	animate(time = 2, pixel_x = -8, easing = CUBIC_EASING)
+	animate(time =  1, pixel_x = 0, easing = CUBIC_EASING | EASE_IN)
+	StartCooldown()
+
+///spawns a plumb bunch of bananas imbued with mystical power.
+/datum/action/cooldown/exquisite_bunch
+	name = "Exquisite Bunch"
+	desc = "Pluck your finest bunch of bananas from your head. This bunch is especially nutrious to monkeykind. A gentle tap will trigger an explosive ripening process."
+	icon_icon = 'icons/obj/hydroponics/harvest.dmi'
+	cooldown_time = 60 SECONDS
+	button_icon_state = "banana_bunch"
+	background_icon_state = "bg_nature"
+	///If we are currently activating our ability.
+	var/activating = FALSE
+
+/datum/action/cooldown/exquisite_bunch/Trigger(atom/target)
+	if(activating)
+		return
+	var/bunch_turf = get_step(owner.loc, owner.dir)
+	if(!bunch_turf)
+		return
+	if(!owner.CanReach(bunch_turf) || !isopenturf(bunch_turf))
+		owner.balloon_alert("can't do that here!")
+		return
+	activating = TRUE
+	if(!do_after(owner, 1 SECONDS))
+		activating = FALSE
+		return
+	playsound(owner, 'sound/creatures/clown/hehe.ogg', 100)
+	if(!do_after(owner, 1 SECONDS))
+		activating = FALSE
+		return
+	activating = FALSE
```

Comment:
```suggestion
	playsound(owner, 'sound/creatures/clown/hehe.ogg', 100)
    activating = FALSE
	if(!do_after(owner, 1 SECONDS))
		return
```

Reply:
Setting activating to false before the do_after will allow the user to spam the button and create multiple bunches,